plugins/bil/bil_main.ml

```diff
@@ -34,44 +33,46 @@ let eliminate_dead_code bil =
       | st -> loop (st :: acc) bil in
   loop [] bil
 
-let apply bil =
-  List.fold ~init:bil ~f:(fun bil f -> f bil)
-
-let norml level =
-  if level = 0 then ident
-  else
-    Stmt.normalize ~keep_ites:false ~normalize_exp:false
+let () =
+  Bil.register_pass "normalization" @@
+  Stmt.normalize ~keep_ites:false ~normalize_exp:false
 
-let simpl level =
-  if level = 0 then ident
-  else
-    Bil.fixpoint
-      (fun bil -> Bil.fold_consts bil |> Bil_const_propagation.run |> eliminate_dead_code)
+let () =
+  Bil.register_pass "optimization" @@
+  Bil.fixpoint
+    (fun bil -> Bil.fold_consts bil |> Bil_const_propagation.run |> eliminate_dead_code)
 
-let run norml_level simpl_level =
-  provide_bil_transformation "internal" @@
-  fun bil -> apply bil [
-    norml norml_level;
-    simpl simpl_level;
-  ]
+let run passes =
+  let passes = ["normalization"; "optimization"] @ passes in
+  let is_selected p =
```

Comment:
your passes parameter should have type `Bil.pass list Term.t` and whatever the parameter returns should be used as a user selection.